src/instrumentations/http.ts

```diff
@@ -42,41 +135,55 @@ export function configureHttpInstrumentation(
     span,
     response
   ) => {
-    if (response instanceof ServerResponse) {
-      const spanContext = span.spanContext();
-
-      if (isSpanContextValid(spanContext)) {
-        const { traceFlags, traceId, spanId } = spanContext;
-        const sampled =
-          (traceFlags & TraceFlags.SAMPLED) === TraceFlags.SAMPLED;
-        const flags = sampled ? '01' : '00';
-
-        appendHeader(
-          response,
-          'Access-Control-Expose-Headers',
-          'Server-Timing'
-        );
-        appendHeader(
-          response,
-          'Server-Timing',
-          `traceparent;desc="00-${traceId}-${spanId}-${flags}"`
-        );
-      }
+    if (!(response instanceof ServerResponse)) {
+      return;
     }
+
+    const spanContext = span.spanContext();
+
+    if (!isSpanContextValid(spanContext)) {
+      return;
+    }
+
+    const { traceFlags, traceId, spanId } = spanContext;
+    const sampled = (traceFlags & TraceFlags.SAMPLED) === TraceFlags.SAMPLED;
+    const flags = sampled ? '01' : '00';
+
+    appendHeader(response, 'Access-Control-Expose-Headers', 'Server-Timing');
+    appendHeader(
+      response,
+      'Server-Timing',
+      `traceparent;desc="00-${traceId}-${spanId}-${flags}"`
+    );
   };
 
   let config = instrumentation._getConfig() as HttpInstrumentationConfig;
 
   if (config === undefined) {
-    config = { responseHook };
-  } else if (config.responseHook !== undefined) {
+    config = {};
+  }
+
+  if (config.responseHook === undefined) {
+    config.responseHook = responseHook;
+  } else {
     const original = config.responseHook;
     config.responseHook = function (this: unknown, span, response) {
       responseHook(span, response);
       original.call(this, span, response);
     };
-  } else {
-    config.responseHook = responseHook;
+  }
+
+  if (shouldAddRequestHook(options)) {
+    const requestHook = createHttpRequestHook(options);
+    if (config.requestHook === undefined) {
+      config.requestHook = requestHook;
+    } else {
+      const original = config.requestHook;
+      config.requestHook = function (this: unknown, span, request) {
+        requestHook(span, request);
+        original.call(this, span, request);
```

Comment:
Instead of always forcing the user to use ours, let's allow them to overwrite it if they need it instead.
See https://github.com/signalfx/splunk-otel-js/pull/312/files#diff-a41c4d0de1c425efb1ab1345aaadeaedc505a09298f326598d0124cce7296f89 for the idea.

Reply:
Not sure about this, if we do this we leave out the possibility to add any customization points if we'd ever need access to the hooks.

Reply:
If I understood you correctly we can still have customization points the same way, which makes me think I probably didn't.
Can you ellaborate?

Reply:
For example if users have configured URL parameter capturing while also using a list of explicit instrumentations with their own hook:
```typescript
startTracing({
  captureRequestUriPrameters: ['sortBy'],
  instrumentations: [
    new HttpInstrumentation({
      requestHook: (span, req) => span.setAttribute('foo', process(req))
    })
  ]
})
```

The main issue is that `captureRequestUriPrameters` will be ignored unless they incorporate our `defaultRequestHook` into their hook. How should they know to do that?

Reply:
I think we're clear now.

By documentation. If they're already fiddling with the advanced configuration, they must've read **some** docs. That's our opportunity to explain the tradeoffs.

As opposed to an advanced user not wanting the performance tradeoff or the additional attributes and not being able to turn it off at all. Since latter("this is not possible, sorry") is unrealistic to be a solution for forever, we eventually would need to introduce another env var `NO_REALLY_DO_NOT_ADD_REQ_HOOK=true`.

So we're talking about the tradeoffs between more (straightforward) control and being opinionated here. I'm leaning towards the former because that leaves us options to do what you suggest later, once we're convinced it's the most common usecase.

Reply:
Sane defaults over magic! :tada:

Reply:
But is it a sane default if you, as an user, have to know which hooks we are using internally? I think the real problem is that HTTP instrumentation allows to register only one hook not multiple.